workflow/controller/operator.go

```diff
@@ -2936,6 +2940,11 @@ func (woc *wfOperationCtx) loadExecutionSpec() (wfv1.TemplateReferenceHolder, wf
 		if woc.controller.Config.WorkflowRestrictions.MustUseReference() {
 			return nil, executionParameters, fmt.Errorf("workflows must use workflowTemplateRef to be executed when the controller is in reference mode")
 		}
+		// Set the WorkflowDefaults from Configmap
+		err := woc.controller.setWorkflowDefaults(woc.wf)
```

Comment:
moved SetWorkflowDefault here to handle all execution-related operations in one place and avoid the validation failure in 'WorkflowTemplateRef` scenario.